src/qspi.rs

```diff
@@ -0,0 +1,548 @@
+    OneBit,
+
+    /// Two IO lines (IO0 and IO1) are used for transmit/receive.
+    TwoBit,
+
+    /// All four IO lines are used for transmit/receive.
+    FourBit,
+}
+
+/// Indicates an error with the QSPI peripheral.
+#[derive(Debug, Copy, Clone)]
+pub enum QspiError {
+    Busy,
+    Underflow,
+    FifoData,
+}
+
+/// Indicates a specific QSPI bank to use.
+#[derive(Debug, Copy, Clone)]
+pub enum Bank {
+    One,
+    Two,
+    Dual,
+}
+
+/// Used to indicate that an IO pin is not used by the QSPI interface.
+pub struct NoIo {}
+
+/// Indicates a set of pins can be used for the QSPI interface on bank 1.
+pub trait PinsBank1 {}
+pub trait PinIo0Bank1 {}
+pub trait PinIo1Bank1 {}
+pub trait PinIo2Bank1 {}
+pub trait PinIo3Bank1 {}
+
+/// Indicates a set of pins can be used for the QSPI interface on bank 2.
+pub trait PinsBank2 {}
+pub trait PinSckBank2 {}
+pub trait PinIo0Bank2 {}
+pub trait PinIo1Bank2 {}
+pub trait PinIo2Bank2 {}
+pub trait PinIo3Bank2 {}
+
+pub trait PinSck {}
+
+impl<SCK, IO0, IO1, IO2, IO3> PinsBank1 for (SCK, IO0, IO1, IO2, IO3)
+where
+    SCK: PinSck,
+    IO0: PinIo0Bank1,
+    IO1: PinIo1Bank1,
+    IO2: PinIo2Bank1,
+    IO3: PinIo3Bank1,
+{
+}
+
+impl<SCK, IO0, IO1, IO2, IO3> PinsBank2 for (SCK, IO0, IO1, IO2, IO3)
+where
+    SCK: PinSck,
+    IO0: PinIo0Bank2,
+    IO1: PinIo1Bank2,
+    IO2: PinIo2Bank2,
+    IO3: PinIo3Bank2,
+{
+}
+
+macro_rules! pins {
+    (Bank1: [IO0: [$($IO0:ty),*] IO1: [$($IO1:ty),*] IO2: [$($IO2:ty),*] IO3: [$($IO3:ty),*]]) => {
+        $(
+            impl PinIo0Bank1 for $IO0 {}
+        )*
+        $(
+            impl PinIo1Bank1 for $IO1 {}
+        )*
+        $(
+            impl PinIo2Bank1 for $IO2 {}
+        )*
+        $(
+            impl PinIo3Bank1 for $IO3 {}
+        )*
+    };
+
+    (Bank2: [IO0: [$($IO0:ty),*] IO1: [$($IO1:ty),*] IO2: [$($IO2:ty),*] IO3: [$($IO3:ty),*]]) => {
+        $(
+            impl PinIo0Bank2 for $IO0 {}
+        )*
+        $(
+            impl PinIo1Bank2 for $IO1 {}
+        )*
+        $(
+            impl PinIo2Bank2 for $IO2 {}
+        )*
+        $(
+            impl PinIo3Bank2 for $IO3 {}
+        )*
+    };
+
+    (SCK: [$($SCK:ty),*], Bank1: $bank1:tt, Bank2: $bank2:tt) => {
+        $(
+            impl PinSck for $SCK {}
+        )*
+        pins!(Bank1: $bank1);
+        pins!(Bank2: $bank2);
+    };
+}
+
+pins! {
+    SCK: [
+        PB2<Alternate<AF9>>,
+        PF10<Alternate<AF9>>
+    ],
+    Bank1: [
+        IO0: [
+            PC9<Alternate<AF9>>,
+            PD11<Alternate<AF9>>
+        ]
+        IO1: [
+            PC10<Alternate<AF9>>,
+            PD12<Alternate<AF9>>,
+            NoIo
+        ]
+        IO2: [
+            PE2<Alternate<AF9>>,
+            PF7<Alternate<AF9>>,
+            NoIo
+        ]
+        IO3: [
+            PA1<Alternate<AF9>>,
+            PD13<Alternate<AF9>>,
+            PF6<Alternate<AF9>>,
+            NoIo
+        ]
+    ],
```

Comment:
This is a rather tedious task.. But the XML files in [modm-devices](https://github.com/modm-io/modm-devices/tree/develop/devices/stm32) are great for cross-checking.

Some corrections

Bank 1
``` 
        IO0: [
            PC9<Alternate<AF9>>,
            PD11<Alternate<AF9>>,
            PF8<Alternate<AF10>>
        ]
        IO1: [
            PC10<Alternate<AF9>>,
            PD12<Alternate<AF9>>,
            PF9<Alternate<AF10>>
        ]
```

Bank 2
```
        IO0: [
            PE7<Alternate<AF10>>,
            PH2<Alternate<AF9>>
        ]
        IO1: [
            PE8<Alternate<AF10>>,
            PH3<Alternate<AF9>>
        ]
```

Reply:
Great catch. I was using the STM32H74x series datasheet for this and forgot to go back and look at the 75x series. Fix incoming.

Reply:
Fixed in 172a69b2eed35031ac014e7760bbc69c4eead8c6#diff-7a546c5c633c8b017f7bb8fa952164fb